pkg/logs/tailers/file/tailer.go

```diff
@@ -344,7 +344,14 @@ func (t *Tailer) forwardMessages() {
 		origin := message.NewOrigin(t.file.Source.UnderlyingSource())
 		origin.Identifier = identifier
 		origin.Offset = strconv.FormatInt(offset, 10)
-		origin.SetTags(append(append(t.tags, t.tagProvider.GetTags()...), output.ParsingExtra.Tags...))
+		tags := append(t.tags, t.tagProvider.GetTags()...)
```

Comment:
Is it safe to reuse `t.tags` here? (I understand that the old code was doing already). 

I may be mistaken, but if there is enough spare capacity in the `t.tags` slice, the memory would be reused and shared between messages, potentially leaking tags from one message to another. It might have been a non-issue if `GetTags` returns the same value for every message, but it may become one now that we have a tag that varies per message.

Reply:
Good point. I see the potential risk here. I'll update this code to copy the `t.tags` slice before assembling the final tag set.